### PR TITLE
docs: add a tf.layers-in-Estimator example [DET-4261]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
We sometimes encounter folks looking to run low-level TF code in our
platform.  We don't support that directly anymore; TF recommends
wrapping low-level graphs in higher-level APIs.

The mnist_tp_to_estimator examples used to show this use case, but since
we removed Tensorpack entirely, we needed a new example.

### CODE REVIEW

**Review by aaron276h — COMMENTED** [review]
Looks good, can we add some tests for this?

File: examples/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
+| Computer Vision | <p>[cifar10_pytorch](computer_vision/cifar10_pytorch)<br>[mnist_multi_output_pytorch](computer_vision/mnist_multi_output_pytorch)<br>[fasterrcnn_coco_pytorch](computer_vision/fasterrcnn_coco_pytorch)<br>[mnist_estimator](computer_vision/mnist_estimator)<br>[iris_tf_keras](computer_vision/iris_tf_keras)<br>[unets_tf_keras](computer_vision/unets_tf_keras)<br>[mnist_tf_layers](computer_vision/mnist_tf_layers)</p> | <p>CIFAR-10<br>MNIST<br>Penn-Fudan Dataset<br>MNIST<br>CIFAR-10<br>Iris Dataset<br>Oxford-IIIT Pet Dataset<br>MNIST</p> | <p>PyTorch<br>PyTorch<br>PyTorch<br>TensorFlow (Estimator API)<br>TensorFlow (tf.keras)<br>TensorFlow (tf.keras)<br>TensorFlow (tf.keras)<br>TensorFlow (tf.layers)</p> |
```
Comment (naren-determined) [inline examples/README.md:5]: Should the framework here be TensorFlow (Estimator API) instead of tf.layers since we do not support that. If so, can we move it next to the other Estimator API example `mnist_estimator`?

File: examples/computer_vision/mnist_tf_layers/model_def.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                return tf.estimator.EstimatorSpec(mode, loss=loss, train_op=train_op)
+
+            if mode == tf.estimator.ModeKeys.EVAL:
+                # Build a graph for validation.
+                logits = calculate_logits(self.context.get_hparams(), images, training=False)
+                predictions = calculate_predictions(logits)
+                error = calculate_error(predictions, labels)
+                return tf.estimator.EstimatorSpec(
+                    mode,
+                    loss=tf.constant([0.0], dtype=tf.float32),
```
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply examples/computer_vision/mnist_tf_layers/model_def.py:105]: oh that's a holdover from some other example.  Deleting it now.

File: examples/computer_vision/mnist_tf_layers/model_def.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    def build_estimator(self) -> tf.estimator.Estimator:
+
+        # Define a model_fn which is the magic ingredient for wrapping a tensorflow graph in an
+        # Estimator.  The Estimator training loop will call this function with different modes to
+        # build graphs for either training or validation (or prediction, but that's not used by
+        # Determined).
+        #
+        # Read more at https://www.tensorflow.org/guide/estimator.
+
+        def model_fn(features, mode):
```
[the remark this answers is not in the document]
Comment (aaron276h) [reply examples/computer_vision/mnist_tf_layers/model_def.py:78]: something like this:

```
def make_model_fn(context):
    def model_fn(features, labels, mode):
        ....
     return model_fn
```

File: examples/README.md (hunk 1 above)
Comment (rb-determined-ai) [reply examples/README.md:5]: What about `TensorFlow (tf.layers via Estimator API)` or something?  Calling this an estimator example is really missing the point of the example.

Comment (naren-determined) [reply examples/README.md:5]: Yeah I like that -- can we move it next to the other Estimator example placement wise in the Computer Vision category?

File: examples/computer_vision/mnist_tf_layers/model_def.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply examples/computer_vision/mnist_tf_layers/model_def.py:78]: I had it out but the `build_model_fn` was a bit silly how empty it was, and I wasn't crazy about the extra layer of function definitions.  Given that, do you still want it in a separate function?

No, the Estimator API actually introspects your function signature.  so you can ask for `labels` and it will be None or you can not ask for it and it's fine.

File: examples/README.md (hunk 1 above)
Comment (rb-determined-ai) [reply examples/README.md:5]: sure

File: examples/computer_vision/mnist_tf_layers/model_def.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaron276h) [reply examples/computer_vision/mnist_tf_layers/model_def.py:78]: I've seen some weird errors where Estimators will make full copy of the objects, not sure if it will apply here but I think it is generally good practice to have users keep it separate.

[the remark this answers is not in the document]
Comment (rb-determined-ai) [reply examples/computer_vision/mnist_tf_layers/model_def.py:78]: ok, done.

**Review by naren-determined — APPROVED** [review]
Looks good!

**Review by aaron276h — APPROVED** [review]
Great work
